javatests/arcs/core/storage/handle/HandleManagerTest.kt

```diff
@@ -111,21 +108,27 @@ class HandleManagerTest {
         assertThat(readBack.isAlive(coroutineContext)).isFalse()
         assertThat(readBack.isDead(coroutineContext)).isTrue()
 
-        // Stash entity1 in the RamDisk manually, so it becomes alive.
-        RamDisk.memory[readBack.storageKey.childKeyWithComponent(readBack.id)] = VolatileEntry(
-            CrdtEntity.Data(VersionMap("foo" to 1), entity1) {
-                if (it is Reference) it
-                else CrdtEntity.Reference.buildReference(it)
-            },
-            0
-        )
+        // Now write the entity via a different handle
+        val singletonEntityHandle = hm.rawEntitySingletonHandle(singletonKey, schema, "entHandle")
+        singletonEntityHandle.store(entity1)
 
         // Reference should be alive.
         assertThat(readBack.isAlive(coroutineContext)).isTrue()
         assertThat(readBack.isDead(coroutineContext)).isFalse()
 
         // Now dereference our read-back reference.
         assertThat(readBack.dereference(coroutineContext)).isEqualTo(entity1)
+
+        singletonEntityHandle.store(entity2)
+
+        // Reference should still be alive.
+        assertThat(readBack.isAlive(coroutineContext)).isTrue()
+        assertThat(readBack.isDead(coroutineContext)).isFalse()
+
+        /* i think this should be true, but it currently fails. What's the expectation here?
```

Comment:
If a reference comes alive, and then the backing entity changes, should the reference also update to reflect the new entity?

Reply:
IMO The reference shouldn't "update", but the next time it's de-referenced, it makes sense that the backing entity would be different from when it was de-referenced before the change.

Reply:
Ok, so it's correct that this test fails, and I can just remove it. 

Or perhaps add a second deref call and verify that.

Reply:
Wait, no... I misread your comment. I think this test *should* be passing (calling `dereference` again should pull out the newer entity) -- I'll dig into this some more.

Reply:
I'm silly. This doesn't work because entity2 is a different entity ID, so updates a different item in the backing store. I need to test *modifying* entity1. :-)